### ISSUE TITLE
How to change perspective of board?

### ISSUE THREAD

**Issue by goshawk22** [deepmind/open_spiel#472]
Is there a way to change the perspective of a state.
For example, in Chess, is it possible to make the white player black and vice versa?

**Comment by lanctot** [comment]
Hi @goshawk22, there is no way to do this currently. 

Hmm.. I wonder if we could add a new game transform that would allow this. I am not sure that we can do it in genersl without game-specific implementations, I would have to think about it more.

Can you describe the use case for more context?

**Comment by goshawk22** [comment]
It would just be a simple function to change the pieces to the opposite colour. For example, in TicTacToe:
XOX           OXO
OXO   -->  XOX
XOX           OXO

**Comment by goshawk22** [comment]
It would just be a simple function to change the pieces to the opposite colour. For example, in TicTacToe:
XOX  
OXO 
XOX
would change to
OXO
XOX
OXO

**Comment by lanctot** [comment]
Yep I understand what you are asking for. I am asking about what you need it for. This will help determine the best way to implement such a feature, as there are several options to choose from.

**Comment by michalsustr** [comment]
I think for Chess you can hack it around by exporting FEN of the current state, swapping the players, and use the updated FEN to create a new initial state.

**Comment by goshawk22** [comment]
I was hoping to use it to generate more training data. In many games, the board could be flipped/rotated. In TicTacToe, there are 4 symmetries and four different states can be identical. This could be used to generate more training data for a model.

**Comment by lanctot** [comment]
> I was hoping to use it to generate more training data. In many games, the board could be flipped/rotated. In TicTacToe, there are 4 symmetries and four different states can be identical. This could be used to generate more training data for a model.

Ah yes, this would be useful. I am thinking one way to do this would be to have a `State::CanonicalSymmetries` function that returns a list/set of states that are either guaranteed to have the same value function under any set of policies (e.g. horizontal / vertical flips and rotations) along with the same set with inverted player perspectives. These would not be (necessarily) legal states but could be used to get observations and augment the data set.

I will discuss the design with the team. Is there one game you're focusing on? Assuming we go with this design, I can probably provide the implementation for 1-2 games and leave the rest to you or the community contributions.

**Comment by goshawk22** [comment]
Yes this would be good. I was primarily focusing on Chess, but I think that this would be useful for all games.
